### PR TITLE
Fix wrong block generation conductor tests

### PR DESCRIPTION
**Pull request by fabioDMFerreira**
* add conductor directive that allows setting magic blocks for different tests. This directive overrides the magic block configuration file specified in `0chain.yaml`.

* Generate new DKG files for the magic block `miners_3_sharder_1`.

* Change env variables names used to find keys and dkg files.

* Separate conductor client initialization from the node registration.

### CODE REVIEW

File: code/go/0chain.net/conductor/conductrpc/client.go [hunk 1]
```diff
@@ -212,3 +212,11 @@ func (c *client) addBlockClientStats(req []byte) (err error) {
 	}
 	return
 }
+
+func (c *client) magicBlock() (configFile *string, err error) {
+	err = c.client.Call("Server.MagicBlock", &struct{}{}, &configFile)
+	for err == rpc.ErrShutdown {
```
Comment (dabasov) [inline code/go/0chain.net/conductor/conductrpc/client.go:218]: what's the reason for this loop?

File: code/go/0chain.net/conductor/conductrpc/entity.go [hunk 2]
```diff
@@ -29,6 +29,18 @@ func (e *Entity) State() (state *State) {
...
 }
 
+// MagicBlock returns the location path of the magic block configuration.
+func (e *Entity) MagicBlock() (magicBlock string) {
+	magicBlockPtr, err := e.client.magicBlock()
+	if err != nil {
+		log.Fatalf("failed getting magic block: %v", err)
+	}
+
+	magicBlock = *magicBlockPtr
```
Comment (dabasov) [inline code/go/0chain.net/conductor/conductrpc/entity.go]: if return unnamed parameter you don't need this line

File: code/go/0chain.net/conductor/conductrpc/server.go [hunk 3]
```diff
@@ -392,6 +394,12 @@ func (s *Server) ShareOrSignsShares(soss *ShareOrSignsSharesEvent,
 	return
 }
 
+// magic block handler
+func (s *Server) MagicBlock(_ *struct{}, configFile *string) (err error) {
+	(*configFile) = s.magicBlock
```
Comment (dabasov) [inline code/go/0chain.net/conductor/conductrpc/server.go:399]: it doesn't make any sense, we can't pass entities by reference

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/conductor/conductrpc/entity.go (hunk 2 above)
Comment (fabioDMFerreira) [reply code/go/0chain.net/conductor/conductrpc/entity.go]: What do you mean?

File: code/go/0chain.net/conductor/conductrpc/server.go (hunk 3 above)
Comment (fabioDMFerreira) [reply code/go/0chain.net/conductor/conductrpc/server.go:399]: I am following the same convention as the other similar methods. What do you suggest doing?

File: code/go/0chain.net/conductor/conductrpc/client.go (hunk 1 above)
Comment (fabioDMFerreira) [reply code/go/0chain.net/conductor/conductrpc/client.go:218]: This came from the state method that uses a polling strategy to fetch state data. Prob. the client connection drops after some time if it is not used. I couldn't find anything related to this on the web though.

Comment (dabasov) [reply code/go/0chain.net/conductor/conductrpc/client.go:218]: hm, weird, usually we want to resend requests every small period of time, not in the loop, but it maybe ok for now

File: code/go/0chain.net/conductor/conductrpc/entity.go (hunk 2 above)
Comment (dabasov) [reply code/go/0chain.net/conductor/conductrpc/entity.go]:
```go
func (e *Entity) MagicBlock() string {
	magicBlockPtr, err := e.client.magicBlock()
	if err != nil {
		log.Fatalf("failed getting magic block: %v", err)
	}

	return *magicBlockPtr
}
```

File: code/go/0chain.net/conductor/conductrpc/server.go (hunk 3 above)
Comment (dabasov) [reply code/go/0chain.net/conductor/conductrpc/server.go:399]: ok, It works somehow and we change the value of *configFile here, it looks weird to be honest, made test here
https://go.dev/play/p/e_N_4lHLk8u

File: code/go/0chain.net/conductor/conductrpc/entity.go (hunk 2 above)
Comment (fabioDMFerreira) [reply code/go/0chain.net/conductor/conductrpc/entity.go]: I got to the named parameters that forgot the other way. Thanks

Comment (fabioDMFerreira) [reply code/go/0chain.net/conductor/conductrpc/entity.go]: fixed by [4922e7c](https://github.com/0chain/0chain/pull/1759/commits/4922e7cb226587a2e44f518f693f73ffc5a0d1b2)

**Review by dabasov — APPROVED** [review]
lgtm

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**